sky/provision/common.py

```diff
@@ -204,8 +204,15 @@ def _get_ips(self, use_internal_ips: bool) -> List[str]:
         return ip_list
 
     def get_feasible_ips(self, force_internal_ips: bool = False) -> List[str]:
-        """Get external IPs if they exist, otherwise get internal ones."""
-        return self._get_ips(not self.has_external_ips() or force_internal_ips)
+        """Get internal IPs if provider config specifies to, external IPs if
+        they exist, otherwise get internal ones."""
```

Comment:
```suggestion
        """Get internal or external IPs depends on the settings."""
```

With this modification, we can leave out the `use_internal_ips` setting here as well?
https://github.com/skypilot-org/skypilot/blob/51f1f78d8b45beaad2f89a9fb0fca2ca03350621/sky/backends/cloud_vm_ray_backend.py#L2251-L2253